Tests/Unit/NetworkServer/BasicsStation/BasicsStationNetworkServerStartupTests.cs

```diff
@@ -27,5 +30,64 @@ public void All_Dependencies_Are_Registered_Correctly()
                 ValidateScopes = true
             });
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ModuleConnectionHostIsInjectedOrNot(bool cloud_deployment)
+        {
+            // arrange
+            Environment.SetEnvironmentVariable("CLOUD_DEPLOYMENT", cloud_deployment.ToString());
+            var services = new ServiceCollection();
+            var config = new ConfigurationBuilder().Build();
+
+            // act + assert
+            var startup = new BasicsStationNetworkServerStartup(config);
+            startup.ConfigureServices(services);
+
+            var serviceProvider = services.BuildServiceProvider(new ServiceProviderOptions
+            {
+                ValidateOnBuild = true,
+                ValidateScopes = true
+            });
+
+            var result = serviceProvider.GetService<ModuleConnectionHost>();
+            if (cloud_deployment)
+            {
+                Assert.Null(result);
+            }
+            else
+            {
+                Assert.NotNull(result);
+            }
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void EnableGatewayTrue_IoTModuleFalse_IsNotSupported(bool cloud_deployment, bool enable_gateway)
+        {
+            // arrange
+            Environment.SetEnvironmentVariable("CLOUD_DEPLOYMENT", cloud_deployment.ToString());
```

Comment:
Same comment as above about the side effects of using env variables. In addition, we should clean up the environment variables at the end of the test.

Reply:
In addition to this, I would move this test only to [ConfigurationTest.cs](https://github.com/Azure/iotedge-lorawan-starterkit/pull/1698/files#diff-0db7d82aa6361a0cec2585b35c85345c82ceacfc335bc5eba9528d518da19ba9)

It is not affecting startup (injection of different services)